unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -165,6 +165,20 @@ public UTF8String toLowerCase() {
     return UTF8String.fromString(toString().toLowerCase());
   }
 
+  public UTF8String initCap(final byte[] byteArr) {
+    if (byteArr == null) return null;
+    if (byteArr[0] <= 122 && byteArr[0] >= 97) //a-z
+    {
+      byteArr[0] = (byte) (byteArr[0] - 32); //white space 32
+    }
+    for (int i = 1; i < byteArr.length; i++) {
+      if (byteArr[i - 1] == 32 || byteArr[i] <= 122 && byteArr[i] >= 97) {
```

Comment:
Shoudn't this be `byteArr[i - 1] == 32 && byteArr[i] <= 122 && byteArr[i] >= 97`? Do the transformation if there's a small character AND the previous one was a whitespace.